mathics/builtin/drawing/plot.py

```diff
@@ -807,7 +807,11 @@ def to_number(x):
                 multiple_colors = True
 
             if not multiple_colors and not self.never_monochrome:
-                colors = [Expression(SymbolRGBColor, *mpl_colors[0])]
+                colors = [
+                    Expression(
+                        SymbolRGBColor, *(MachineReal(cc) for cc in mpl_colors[0])
```

Comment:
Actually, for this kind of thing, to_expression is better and let me explain why....

WIth Expression, we need to loop over the list to get element properties. So we do two loops. One here and one in Expression. 

Also with to_expression we record the fact that the elements here all come from MachineReal, and so in the future if we want to do the inverse conversion we may have head start here. 

Finally, I do thinks removing the generator loop simplifies things.